src/test/java/org/broadinstitute/hellbender/engine/MultiVariantDataSourceUnitTest.java

```diff
@@ -0,0 +1,145 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.variant.variantcontext.VariantContext;
+
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Iterator;
+
+public final class MultiVariantDataSourceUnitTest extends BaseTest {
```

Comment:
Can you reproduce some of the query test cases from `FeatureDataSourceUnitTest` here?